### ISSUE TITLE
[a11y][Page] Secondary actions should have an ability to receive focus

### ISSUE THREAD

**Issue by tatianau** [Shopify/polaris-react#8105]
## Issue summary

There is a focus management issue prevalent across the Admin. Many of the Page secondary actions in the Admin - e.g. Import, Export, Duplicate, etc. - open a modal, and currently there is no easy way to [return focus to the modal activator](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/#keyboard-interaction-7) when the activator is the secondary action of the Page component that doesn't support it.

## Expected behavior

Ability to connect Page secondary actions to the modal or any other overlay to be able to return focus to the activator. 
For example, by adding the ability to provide an `activatorRef` to `<Page />`'s `secondaryActions` to then pass it to the `<Modal />` as an `activator`.

## Actual behavior
No ability to connect `<Page />`'s secondary actions to the overlay they open.

## Steps to reproduce the problem

1. Open any section that has secondary actions that open a modal.
  1.1. Example actions: Export, Import, Duplicate. 
  1.2. Example pages: Discount index, Product index, Product details, Variant details.
1. Use keyboard tab to navigate through the page
2. Activate the action to open a modal
3. Close modal or cancel out of it
4. See the focus returns to the top of the page instead of the modal activator

<details>
      <summary>Product export modal focus</summary>
      <video src="https://user-images.githubusercontent.com/15054990/188015493-77e0a4ff-d05b-4462-8eff-14e8d6a5c6cc.mp4">
</details>

## Relevant discussions and previous attempts to fix it
https://github.com/Shopify/polaris/issues/8058, https://github.com/Shopify/polaris/pull/5258, https://github.com/Shopify/polaris/pull/5495, https://github.com/Shopify/web-configs/pull/349

**Comment by tatianau** [comment]
@oksanashopify this might be an interesting issue to tackle / pair on with such a broad impact, also cc @jakemoves @stefanlegg @maxariss for context.

**Comment by oksanashopify** [comment]
Posting here just to keep an eye on ongoing conversation about  
[Consolidate "page actions" in Resource detail layout pattern](https://github.com/Shopify/polaris/issues/8058#top)

**Comment by knames** [comment]
https://github.com/Shopify/polaris/pull/8701 will partially solve this. if a secondary action ends up in a rollup/popover, the popover is closed on click so there's nothing to return to. I attempted many times to solve this, but there's weirdness with items popping in and out of the More Actions, resizing issues (and overlapping titles).

**Comment by knames** [comment]
Here's a branch that has popover's that stay open when provided with id's and refs, should anyone want to try and run it the last mile

`ks-popover-stay`

**Comment by tatianau** [comment]
Resolved by https://github.com/Shopify/polaris-internal/pull/2302

**Comment by jakemoves** [comment]
🍾🥂 way to take this all the way @tatianau !
